clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/security/KubernetesV1Credentials.java

```diff
@@ -294,4 +295,32 @@ public Boolean isRegisteredImagePullSecret(String secret, String namespace) {
     }
     return secrets.contains(secret);
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (!(o instanceof KubernetesV1Credentials)) {
+      return false;
+    }
+    KubernetesV1Credentials that = (KubernetesV1Credentials) o;
+    return Objects.equals(namespaces, that.namespaces)
+        && Objects.equals(omitNamespaces, that.omitNamespaces)
+        && Objects.equals(dockerRegistries, that.dockerRegistries)
```

Comment:
These are `List<LinkedDockerRegistryConfiguration>` but `LinkedDockerRegistryConfiguration` does not override `equals` and `hashCode` so the lists will only be equal if they contain the same object references.

Reply:
Ah, yes. I remember attempting to implement `equals` on `LinkedDockerRegistryConfiguration` but encountered some issues. I don't remember what they were exactly, though. Will revisit.